### DISCUSSION TITLE
How to install unocss for a rails 7 app

### DISCUSSION THREAD

**Discussion by benjaminb10** [unocss/unocss#1913] [Q&A]
Hi. I try to install Uno CSS on a new rails 7 app.
I can't find any tutorial for that.
Such as this one for tailwind dans Rails https://www.simplethread.com/how-to-create-a-new-rails-7-app-with-tailwind/

Do you recommand to install it via webpacker?

**Comment by AxelBriche** [comment]
I think you have to install the [cli version of "unocss"](https://www.npmjs.com/package/@unocss/cli), then add the config file `unocss.config.js`.

At this stage you should already be able to generate an `unocss.css` file in command line from your console with something like this:
`unocss app/views/**/*.html.erb -o app/javascript/styles/unocss.css --watch`

_Note: don't change the `/app/javascript/` path, it is for webpack._

You can use this command with more path, example:
`unocss "app/views/**/*.html.erb" "app/helpers/*.rb" -o app/javascript/styles/unocss.css --watch`

Then, add `import '../styles/unocss.css'` inside `app/javascript/packs/application.js`

Finally, add `<%= stylesheet_pack_tag "application", 'data-turbolinks-track': 'reload' %>` inside your layout `app/views/layouts/application.html.erb`

For production, use the same unocss command without `--watch`.
